models/Layer/Layer.js

```diff
@@ -99,6 +99,35 @@ class Layer {
     return this.layers;
   }
 
+  /**
+   * Get all child, grandchild, etc layers, and optionally filter them by a predicate function.
+   * @param {function} [predicate] - Filter function if you want to only return certain layers
+   * @returns {Layer[]} An array of all layers. Will return all direct children, grandchildren, etc.
+   */
+  getAllLayers(predicate) {
+    function getRecursiveLayers(current, list) {
+      if (current === null || current === undefined) {
+        return list;
+      }
+      const childLayers = current.getLayers();
+      if (childLayers !== undefined) {
+        childLayers.forEach(layer => {
+          list.push(layer);
+          getRecursiveLayers(layer, list);
+        });
+      }
+      return list;
+    }
+
+    const allLayers = getRecursiveLayers(this, []);
+
+    if (predicate) {
```

Comment:
This probably won't be an issue now, but in the future we might want to optimize this to only do a single pass so we don't create the full array then filter the array. But I like the readability and cleanliness of this for now, so lets keep it until we get performance issues.